cisagov/ScubaGear
# Updated OneDrive 2.1, 2.2 and 2.3 policy

Modified OneDrive 2.1, 2.2 and 2.3 policy

## 🗣 Description ##

Modified OneDrive 2.2 and 2.3 policy, so that
1. When 2.1 pass, 2.2 and 2.3 automatically pass and indicated "Requirement met: AnyOne Link is disabled"
2. fixed the checks for policy 2.2 
3. 2.1 now pass for more restricted sharing settings

## 💭 Motivation and context ##

closes #31 #127 #154 

## 🧪 Testing ##

1. Open the Sharepoint Admin Center
2. Navigate to Policies > Sharing
3. Set the External Sharing slider to New and existing guests. (run test- 2.1, 2.2, 2.3 pass)
5. Set the External Sharing slider to Anyone. 
6. Scroll to the section "Choose expiration and permissions options for Anyone links." 
To test 2.2 
7. Uncheck "These links must expire within this many days" (run test -2.1 fail, 2.2 fails)
8. Uncheck "These links must expire within this many days" and input number thats not 30 (run test -2.1 fail, 2.2 fails)
9. Uncheck "These links must expire within this many days" and input number 30 (run test -2.1 fail, 2.2 pass)

To test 2.3 (tested in #112)
Mainly just to make sure when 2.1 pass, 2.3 pass

PS: Don't forget to revert the changes after testing

## ✅ Pre-approval checklist ##






      in code comments.
      to reflect the changes in this PR.

## ✅ Pre-merge checklist ##





## ✅ Post-merge checklist ##
